v2/api/dbforpostgresql/customizations/flexible_server_extensions.go

```diff
@@ -79,3 +85,55 @@ func secretsToWrite(obj *postgresql.FlexibleServer) ([]*v1.Secret, error) {
 
 	return collector.Values()
 }
+
+var _ extensions.PreReconciliationChecker = &FlexibleServerExtension{}
+
+// If a flexible has a provisioningState in this set, it will reject any attempt to PUT a new state out of hand;
+// so there's no point in even trying. This is true even if the PUT we're doing will have no effect on the state of the
+// cluster.
+// These are all listed lowercase, so we can do a case-insensitive match.
+var blockingFlexibleServerStates = set.Make(
```

Comment:
TBH it might be better to define this as a set of allowed states, rather than a set of disallowed states. ARM defines only [3 terminal states](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#provisioningstate-property)

Those states are:
- Succeeded
- Failed
- Canceled 

Anything other than those 3 states is a transitionary state and will likely cause request reject (that's how AKS works at least, 99% sure most other RPs with "only 1 request active at a time" are the same)

Reply:
This also makes us more hardened against possible futures where they add new "-ing" states in the future (which they are allowed to do)

Reply:
Fair call. Done.